src/App.tsx

```diff
@@ -1,26 +1,86 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState } from 'react';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { useSelector } from 'react-redux';
+import { RootState } from './app/store';
+import { Status } from './types/Status';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const filterData = useSelector((state: RootState) => state.filter);
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  function getVisibleTodos(
+    allTodos: Todo[],
+    {
+      query,
+      status,
+    }: {
+      query: string;
+      status: Status;
+    },
+  ) {
+    let handledTodos = allTodos;
+
+    if (query) {
+      handledTodos = handledTodos.filter(todo =>
+        todo.title.toLowerCase().includes(query.toLowerCase()),
+      );
+    }
+
+    switch (status) {
+      case 'active':
+        handledTodos = handledTodos.filter(todo => !todo.completed);
+        break;
+      case 'completed':
+        handledTodos = handledTodos.filter(todo => todo.completed);
+        break;
+      default:
+        return handledTodos;
+    }
+
+    return handledTodos;
+  }
```

Comment:
The `getVisibleTodos` function is functional, but consider adding comments or breaking it down into smaller functions for better readability and maintainability. This can help other developers understand the logic more easily.